### ISSUE TITLE
Learning rate preview

### ISSUE THREAD

**Issue by Arilziem** [d8ahazard/sd_dreambooth_extension#724]
It would be nice to have a preview of the lr over the lifetime of the training session in the settings. Currently, it's hard to understand what is going on unless you already have intimate knowledge of the different functions.

This could either be done by generating an image, or by generating an SVG (nicer, could be done with either python or js). Would fit nicely under the learning rate section, maybe with a toggle to show/hide.

Ideally, the data for the preview is generated by the same function that also dictates the lr during training.

A live preview that is fed through the normal lr function would make tuning more intuitive and as a nice side effect can also serve as a sanity check when something with lr calculation is changed or a new lr function is implemented.

**Comment by Arilziem** [comment]
Pretty pictures are always a good idea.

Part of why I'm asking is because I swear cosine annealing behaved differently a few commits ago, now I cannot get it to do multiple "up's and down's" over the scheduled epochs. Now if I want to debug I can either do it with my huge-ass dataset or create a test data set, try out a few things ... That is why I am suggesting a preview. Way faster iteration for users and faster to spot errors if there are any.

I get your hesitance though, I did webdev for a pretty long time and honestly would rather not touch the "web" part of the webui.

**Comment by d8ahazard** [comment]
If you want the multiple ups and downs, then you're looking for cosine_annealing_with_restart

**Comment by Arilziem** [comment]
Yes I am using that one now, the only issue I have with it is that images later on in the dataset always get less weight. I guess I'm looking for a sort of cyclic cosine annealing (well basically just a cyclic cosine), but if that was never a thing I must have hallucinated that.

I still think having a preview would make the learning rate way more accessible, but I guess UX is always more of a nice-to-have compared to things, well, working.

**Comment by d8ahazard** [comment]
I don't have a pretty graph, but the "Debug Buckets" function now includes the learning rate in the file output to "samples".
